### PR TITLE
opencv 4.x: fix android build

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/15228

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I've cherry picked 3 commits of https://github.com/conan-io/conan-center-index/pull/15193 (huge PR), which fix Android.

File: recipes/opencv/4.x/conandata.yml [hunk 1]
```diff
@@ -40,6 +40,12 @@ patches:
     - patch_file: "patches/4.1.2-0003-find-quirc.patch"
       patch_description: "Robust discovery & injection of quirc"
       patch_type: "conan"
+    - patch_file: "patches/4.5.2-0001-fix-zlib-static-android.patch"
+      patch_description: "Fix discovery of zlib static if Android"
+      patch_type: "conan"
```
Comment (prince-chrismc) [inline recipes/opencv/4.x/conandata.yml:45]: Maybe portability? Or is the "system" one just named different then the Conan one?

**Review by prince-chrismc — APPROVED** [review]

**Comment by uilianries** [comment]
I just restarted it again.

**Comment by uilianries** [comment]
Ubuntu server is having a tough day, the APT server is out: https://github.com/community/community/discussions/44171

**Review by jwillikers — APPROVED** [review]

**Pull request merged**